### ISSUE TITLE
missing validations & tests

### ISSUE THREAD

**Issue by albjeremias** [helpbuttons/hb-back#41]
- [x] check if user is owner of network/button/buttonTemplate, before deleting or removing depends on #10 
- [x] add uploads to models, upload should store file with the model indexed
>> the frontend, should upload the file separately, and then use the url to store on the model!
- [x] merge file-download & file-upload controllers
- [x] check ForeignKeys, or write tests failing the FK
- [ ] define friendNetworks api call
- [x]  missing tests for created/modified -> won't do.
- [x] loopback wont validate if username already exists.. maybe need to put as primarykey on db

**Comment by albjeremias** [comment]
on validation using loopback 4
https://github.com/loopbackio/loopback-next/issues/2735#issuecomment-483425305

and documentation here:
https://ajv.js.org/json-schema.html#pattern

```
@model()
class Product extends Entity {
  @property({
    name: 'name',
    description: "The product's common name.",
    type: 'string',
    // Specify the JSON validation rules here
    jsonSchema: {
      maxLength: 30,
      minLength: 10,
    },
  })
  public name: string;
}
```
